cmd/juju/application/bundle/bundle.go

```diff
@@ -345,3 +350,55 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
 	}
 	return errors.Trace(verifyError)
 }
+
+func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
```

Comment:
Suggest you collect the errors and return all of them. It'd be very frustrating to get one error and have 6 more to fix in a large bundle, finding the errors one at a time. This would also allow you to test all three types of mismatch at once.

Reply:
Yes good idea. Done